packages/core/src/common/nls.ts

```diff
@@ -24,7 +24,64 @@ export namespace nls {
 
     export const locale = typeof window === 'object' && window && window.localStorage.getItem(localeId) || undefined;
 
+    let keyProvider: LocalizationKeyProvider | undefined;
+
+    /**
+     * Automatically localizes a text if that text also exists in the vscode repository.
+     */
+    export function localizeByDefault(defaultValue: string, ...args: FormatType[]): string {
+        if (localization) {
+            if (!keyProvider) {
+                keyProvider = new LocalizationKeyProvider();
+            }
+            const key = keyProvider.get(defaultValue);
+            if (key) {
+                return localize(key, defaultValue, ...args);
+            }
+        }
+        return Localization.format(defaultValue, args);
+    }
+
     export function localize(key: string, defaultValue: string, ...args: FormatType[]): string {
         return Localization.localize(localization, key, defaultValue, ...args);
     }
 }
+
+interface NlsKeys {
+    [key: string]: (string | NlsInfo)[]
+}
+
+interface NlsInfo {
+    key: string
+    comment: string[]
+}
+
+class LocalizationKeyProvider {
+
+    private data = this.buildData();
+
+    get(defaultValue: string): string | undefined {
+        return this.data.get(defaultValue);
+    }
+
+    private buildData(): Map<string, string> {
```

Comment:
Would you mind adding a comment / some comments here to indicate what the source data looks like and what the goal is?